### ISSUE TITLE
How does this work with Cloudflare?

### ISSUE THREAD

**Issue by OultimoCoder** [oramasearch/orama#228]
If I have a dataset with 11 million rows and about 1.4Gb of data would this be a good fit? I can't find much information about how it works with workers or anything like that.

**Comment by abhay187** [comment]
It is simply not possible. Cloudflare worker has just 128 MB of memory and a script size of 5 MB. Also, Lyra is not built for every use case. I think it is a good option for small-scale websites that need search or autocomplete.

**Comment by OultimoCoder** [comment]
I heard some rumblings about using kv indexes and using durable objects as persistent storage. has that not been explored yet?

**Comment by abhay187** [comment]
Still, it is impossible because Lyra needs indexed data in memory for search. Indexing 1.4GB of data can easily go beyond 10 GB. Cloudflare workers have only 128 MB of memory.

**Comment by OultimoCoder** [comment]
Quote from cloudflare discord discussing this:

```
yes, exactly!  But I also don't think it has to be limited to 128mb ... I've been poking around in the data structures inside Lyra ... and I think they can be tweaked to use the DO Transactional Storage API ... so the indexes could even be many 10s of GBs

Wow, that would be quite amazing. Would you split the index based on first letters of a prefix?

Exactly - that's how Lyra is storing it ... the index is just a nested btree object ... so if that was flattened, it should be able to use the DO Transactional Storage API to do the lookup instead of just traversing the in-memory tree ... You'd want a flag of course, because storing it in say a few sharded KV docs or a single R2 file if say the index was less than 100mb or so ... then the lookups can actually happen at the worker not the DO
so the other idea I was playing with, is that the first layer or two of the index could actually be in KV ... so then let's say each letter ... or the first 2 letters ... or even the first 3 letters of could be it's own KV document ~25mb ... the sharding gets more complex but it would be much more globally performant because it wouldn't have to go back to the DO ... but the pure DO-based index seems the simplest even if it's not the fastest ...
```

Example of another project which has the whole of wikipedia indexed by edge search:

https://wilsonl.in/wiki/
https://github.com/wilsonzlin/edgesearch

**Comment by abhay187** [comment]
My answer is based on what is currently possible. Sharding database is used in all major search engines such as Elastic search. Managing these shards requires a bit of work.
Edgesearch seems promising but lags behind due to good documentation. Also, I am getting cors errors while testing examples.

**Comment by OultimoCoder** [comment]
But I mean edgesearch proves it's possible. You were wrong to say it's
impossible.

**Comment by abhay187** [comment]
Yeah, kind of because I was not aware of edgesearch. I don't know why it is not promoted as Lyra.

**Comment by abhay187** [comment]
Also making something and building to scale it for production requirements takes years. I Hope Lyra grows to such requirements but as of now, I will prefer a proven solution for production.

@OultimoCoder What stops you from using edgesearch?

**Comment by OultimoCoder** [comment]
No relevancy ranking, I agree the documentation is lacking. But I would like some documents to be returned over others based on word proximity, e.g. search "I have" would return all documents with the terms "I have" but wouldn't rank "I have walked to the park" over "I like to go swimming and have an icecream".

**Comment by abhay187** [comment]
Exactly what I said making something production ready takes years. Right now you can only search in the string field. I will use Lyra for noncritical searches using a small data set of a few MB.

@OultimoCoder, For your use case I would recommend [Meilisearch](https://docs.meilisearch.com/learn/what_is_meilisearch/overview.html). You can deploy it on [Railway.app](https://docs.meilisearch.com/learn/cookbooks/railway.html) and probably run it for 5-10 $.

Try meilisearch on railway.app and let me know if you need help.

**Comment by OultimoCoder** [comment]
Meilisearch isn't production ready though. I already looked at it and it looks great but lots of reports of it crashing unexpectedly, no request queing so can't handle load/high request throughput and only supports vertical scaling.

**Comment by abhay187** [comment]
Ok, I didn't try it for GB of data. Their support team is pretty helpful. If you ask they can probably help you.

**Comment by spirobel** [comment]
there is also typesense. But I agree with you @OultimoCoder  it would be super amazing if there was a way to do sharding.

**Comment by micheleriva** [comment]
Hi @OultimoCoder, @abhay187,
let me cycle back to you on this issue.

We rebranded **Lyra** into **Orama** and founded a company to support its development. Here's the announcement post: https://micheleriva.medium.com/ive-founded-a-company-introducing-oramasearch-inc-f69121b6b1c3

We are working on a solution to make Orama run with large volumes of data on Cloudflare. It's doable, and we have opened a private beta for it, in case you're interested in joining: https://orama.to/private-beta-signup

I'd also suggest you to join our Slack channel, where we can help you integrate Orama in your existing stack: https://orama.to/slack
